### PR TITLE
Fix XcodeDeps includes skipped dependencies

### LINKED ISSUE #13818: [bug] XcodeDeps includes skipped dependencies
### Environment details

* Operating System+version: *macOS 13.3.1*
* Compiler+version: *clang 14.0.3*
* Conan version: *2.0.4*
* Python version: *3.9*

### Steps to reproduce

1. Run `conan install --output-folder ./generated -g XcodeDeps --requires=tensorflow-lite/2.10.0 -s compiler.cppstd=17 -s build_type=Release`
2. `generated/conan_tensorflow_lite_tensorflow_lite.xcconfig` now contains an include directive for non-existent file `conan_eigen.xcconfig`. The same happens for skipped dependency `fxdiv`.

### Logs

```
> conan install --output-folder ./generated -g XcodeDeps --requires=tensorflow-lite/2.10.0 -s compiler.cppstd=17 -s build_type=Release

======== Input profiles ========
Profile host:
[settings]
arch=armv8
build_type=Release
compiler=apple-clang
compiler.cppstd=17
compiler.libcxx=libc++
compiler.version=14
os=Macos
[conf]


Profile build:
[settings]
arch=armv8
build_type=Release
compiler=apple-clang
compiler.cppstd=gnu17
compiler.libcxx=libc++
compiler.version=14
os=Macos
[conf]



======== Computing dependency graph ========
flatbuffers/23.1.21: RUN: cmake --version

Graph root
    cli
Requirements
    abseil/20230125.1#9037026f554e4b26acdb80576838fd38 - Cache
    cpuinfo/cci.20220228#120d41d8c7b6070d8a065873ecce3ba3 - Cache
    eigen/3.4.0#b199eacbe6a07d9a01f1bc4a1995acc4 - Cache
    farmhash/cci.20190513#87016cdc5c7f2599ae29e31f1f47666f - Cache
    fft/cci.20061228#1d323fe06047bbe4c66142372c209f88 - Cache
    flatbuffers/23.1.21#6a9f109afe4823ff3c98a93d6a82396c - Cache
    fp16/cci.20210320#80b2c477ce3144bdedbf85bf53c82758 - Cache
    fxdiv/cci.20200417#cd81cfae19f7e26153aad6323f53bb89 - Cache
    gemmlowp/cci.20210928#7237eaa845b4c3b1b81742c292528778 - Cache
    psimd/cci.20200517#a30c8d46ee434b677448ed267ed012c5 - Cache
    pthreadpool/cci.20210218#ba465db5167833fa1a243e432544ee90 - Cache
    ruy/cci.20220628#e0cea2739a40ab73ebe87820437fceb1 - Cache
    tensorflow-lite/2.10.0#adf57fd97aae5d15be8559aab5cd473a - Cache
    xnnpack/cci.20220801#bfa9879ec27cd824343c2e45e6fc691c - Cache
Build requirements
    cmake/3.25.3#51f407c176f55f883d98d74e77de824e - Cache

======== Computing necessary packages ========
Requirements
    abseil/20230125.1#9037026f554e4b26acdb80576838fd38:d8b3817cfaed643e7901be467919a05da49a8b19#8cb315d115d985896c3cc4880fe03fd1 - Cache
    cpuinfo/cci.20220228#120d41d8c7b6070d8a065873ecce3ba3:ff16f049d4e0f54cec2bab74c4f01c86f7f5acfa#f5b93d821cf1aabfa80ccf027fcd9215 - Cache
    eigen/3.4.0#b199eacbe6a07d9a01f1bc4a1995acc4:da39a3ee5e6b4b0d3255bfef95601890afd80709#b2e7c2d86c5d1dbefc534889aa72e12c - Skip
    farmhash/cci.20190513#87016cdc5c7f2599ae29e31f1f47666f:42d1d40ae88c7bf9c7222e1cd3e3301c4248db2f#5d29343743c9adf2751a41c334f3a976 - Cache
    fft/cci.20061228#1d323fe06047bbe4c66142372c209f88:240c2fad2eab8492ecbc70baa3e4f9ccea407a8c#2d0fea65b57159b70e7f217c19f6e9f1 - Cache
    flatbuffers/23.1.21#6a9f109afe4823ff3c98a93d6a82396c:60ad6fd783a9455d15c6530e0cf0dba10214c3ec#fe6215bf31fce23d8d70ed8d0932bcae - Cache
    fp16/cci.20210320#80b2c477ce3144bdedbf85bf53c82758:da39a3ee5e6b4b0d3255bfef95601890afd80709#b7ab18bd2ee17d82530f93d190f36156 - Cache
    fxdiv/cci.20200417#cd81cfae19f7e26153aad6323f53bb89:da39a3ee5e6b4b0d3255bfef95601890afd80709#99822879bdd86a69aa08623a42893b48 - Skip
    gemmlowp/cci.20210928#7237eaa845b4c3b1b81742c292528778:d8b3817cfaed643e7901be467919a05da49a8b19#c81a87f2cd36fa267a70ca3496af5470 - Cache
    psimd/cci.20200517#a30c8d46ee434b677448ed267ed012c5:da39a3ee5e6b4b0d3255bfef95601890afd80709#d8c3d8d76275b4c8abf8ee833e49cae5 - Cache
    pthreadpool/cci.20210218#ba465db5167833fa1a243e432544ee90:465447c888cd6f493ad17afbf4f59030e62e1d72#b7277fa60c48c5d8b109c493bc1b6b09 - Cache
    ruy/cci.20220628#e0cea2739a40ab73ebe87820437fceb1:719c8f68688a1e9f6c9a78030a281db1c880aa6e#81a242a810d0ee1cbf3d6d0a0f3aff56 - Cache
    tensorflow-lite/2.10.0#adf57fd97aae5d15be8559aab5cd473a:8eb3f64416693659bd23f85899bd7dbe3923ef68#f11e86949426e96e15974f3a4055bc8d - Cache
    xnnpack/cci.20220801#bfa9879ec27cd824343c2e45e6fc691c:db3f072c4a3f96df73df977a98c5185b08b8fbd0#b8581d342b4e8a004a5bc579767ed0ee - Cache
Build requirements
    cmake/3.25.3#51f407c176f55f883d98d74e77de824e:9e5323c65b94ae38c3c733fe12637776db0119a5#a97e3811d3949be07a083f1b9d9d3fa4 - Skip

======== Installing packages ========
abseil/20230125.1: Already installed! (1 of 12)
cpuinfo/cci.20220228: Already installed! (2 of 12)
farmhash/cci.20190513: Already installed! (3 of 12)
fft/cci.20061228: Already installed! (4 of 12)
flatbuffers/23.1.21: Already installed! (5 of 12)
gemmlowp/cci.20210928: Already installed! (6 of 12)
psimd/cci.20200517: Already installed! (7 of 12)
pthreadpool/cci.20210218: Already installed! (8 of 12)
fp16/cci.20210320: Already installed! (9 of 12)
ruy/cci.20220628: Already installed! (10 of 12)
xnnpack/cci.20220801: Already installed! (11 of 12)
tensorflow-lite/2.10.0: Already installed! (12 of 12)
WARN: Usage of deprecated Conan 1.X features that will be removed in Conan 2.X:
WARN:     'cpp_info.names' used in: abseil/20230125.1, tensorflow-lite/2.10.0, flatbuffers/23.1.21
WARN:     'cpp_info.build_modules' used in: abseil/20230125.1, tensorflow-lite/2.10.0, flatbuffers/23.1.21
WARN:     'cpp_info.filenames' used in: flatbuffers/23.1.21
WARN:     'env_info' used in: flatbuffers/23.1.21

======== Finalizing install (deploy, generators) ========
cli: Writing generators to /Users/kim.lindberger/cotton/build/generated
cli: Generator 'XcodeDeps' calling 'generate()'
cli: Generating aggregated env files
cli: Generated aggregated env files: ['conanbuild.sh', 'conanrun.sh']
Install finished successfully
```

### PR DESCRIPTION
**Pull request by czoido**
Changelog: Fix: Fix XcodeDeps includes skipped dependencies.
Docs: omit

Closes: https://github.com/conan-io/conan/issues/13818

### CODE REVIEW

File: conan/tools/apple/xcodedeps.py [hunk 1]
```diff
@@ -206,9 +206,10 @@ def _all_xconfig_file(self, deps, content):
         """
         content_multi = content or self._all_xconfig
 
-        for req, dep in deps.items():
-            dep_name = _format_name(dep.ref.name)
-            content_multi = content_multi + '\n#include "conan_{}.xcconfig"\n'.format(dep_name)
+        for dep in deps.values():
```
Comment (czoido) [inline conan/tools/apple/xcodedeps.py:209]: Not related with the issue, I found that some includes were repeated

File: conan/tools/apple/xcodedeps.py [hunk 2]
```diff
@@ -206,9 +206,10 @@ def _all_xconfig_file(self, deps, content):
...
+            include_file = f'conan_{_format_name(dep.ref.name)}.xcconfig'
+            if include_file not in content_multi:
```
Comment (czoido) [inline conan/tools/apple/xcodedeps.py:211]: Not related with the issue, I found that sometimes some includes were repeated. This fixes that

File: conan/tools/apple/xcodedeps.py [hunk 3]
```diff
@@ -131,7 +131,7 @@ def _conf_xconfig_file(self, require, pkg_name, comp_name, package_folder, trans
         """
         def _merged_vars(name):
             merged = [var for cpp_info in transitive_cpp_infos for var in getattr(cpp_info, name)]
-            return list(OrderedDict.fromkeys(merged).keys())
```
Comment (memsharded) [inline conan/tools/apple/xcodedeps.py]: wasn't this to preserve order of merged things?

Comment (czoido) [reply conan/tools/apple/xcodedeps.py]: Yes, let me revert that

File: conan/tools/apple/xcodedeps.py [hunk 4]
```diff
@@ -131,7 +131,7 @@ def _conf_xconfig_file(self, require, pkg_name, comp_name, package_folder, trans
...
+            return list(set(merged))
```
Comment (czoido) [inline conan/tools/apple/xcodedeps.py]:
```suggestion
            return list(OrderedDict.fromkeys(merged).keys())
```

File: conan/tools/apple/xcodedeps.py [hunk 5]
```diff
@@ -288,8 +289,8 @@ def _transitive_components(component):
                     _transitive_components(comp_cpp_info)
 
                     # remove duplicates
-                    transitive_internal = list(OrderedDict.fromkeys(transitive_internal).keys())
-                    transitive_external = list(OrderedDict.fromkeys(transitive_external).keys())
+                    transitive_internal = list(set(transitive_internal))
+                    transitive_external = list(set(transitive_external))
```
Comment (czoido) [inline conan/tools/apple/xcodedeps.py]:
```suggestion
                    transitive_internal = list(OrderedDict.fromkeys(transitive_internal).keys())
                    transitive_external = list(OrderedDict.fromkeys(transitive_external).keys())
```

**Review by memsharded — APPROVED** [review]

**Pull request merged**